### PR TITLE
type_to_default patch for long double

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
Now creates a temp var that is initialized then assigned to the member.  Context is needed for non unions to prevent multiple declarations.

### CODE REVIEW

File: dds/idl/dds_generator.cpp [hunk 1]
```diff
@@ -332,11 +332,13 @@ string type_to_default(const std::string& indent, AST_Type* type, const string&
       if (use_cxx11) {
         def_val = "0.0L";
       } else {
+		    string temp_val = "ACE_CDR::LongDouble val = ACE_CDR_LONG_DOUBLE_INITIALIZER";
```
Comment (mitza-oci) [inline dds/idl/dds_generator.cpp:335]: Indent looks wrong... tabs?

File: dds/idl/dds_generator.cpp [hunk 2]
```diff
@@ -332,11 +332,13 @@ string type_to_default(const std::string& indent, AST_Type* type, const string&
...
         if (is_union) {
-          def_val = "ACE_CDR_LONG_DOUBLE_INITIALIZER";
+          def_val = "val";
+		  return indent + temp_val + ";\n" + indent + name + pre + def_val + post + ";\n";
```
Comment (mitza-oci) [inline dds/idl/dds_generator.cpp:338]: Indent looks wrong... tabs?

File: dds/idl/dds_generator.cpp [hunk 3]
```diff
@@ -332,11 +332,13 @@ string type_to_default(const std::string& indent, AST_Type* type, const string&
...
         } else {
-          def_val = "ACE_CDR_LONG_DOUBLE_ASSIGNMENT(" + name + ", 0)";
-          return indent + def_val + ";\n";
+          def_val += "ACE_CDR_LONG_DOUBLE_ASSIGNMENT(" + name + ", val)";
+          return indent + "{\n  " + indent + temp_val + ";\n  " + indent + def_val + ";\n" + indent + "}\n";
```
Comment (mitza-oci) [inline dds/idl/dds_generator.cpp:341]:
```suggestion
          return indent + "{\n"
            "  " + indent + temp_val + ";\n"
            "  " + indent + def_val + ";\n" + 
            indent + "}\n";
```

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: dds/idl/dds_generator.cpp (hunk 1 above)
Comment (ClaytonCalabrese) [reply dds/idl/dds_generator.cpp:335]: It doesn't look like this for me. I don't know how this is happening on github as it isn't what is on my branch.

Comment (ClaytonCalabrese) [reply dds/idl/dds_generator.cpp:335]: Nevermind, you are right there were tabs which were interpreted differently locally than on github.  Weird.

Comment (iguessthislldo) [reply dds/idl/dds_generator.cpp:335]: Which I think is the main reason for not using tabs. Tabs are traditionally 8 spaces which Github is doing, but editors can be configured to display them differently.

Comment (ClaytonCalabrese) [reply dds/idl/dds_generator.cpp]: Okay, thank you.

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**